addons/road-generator/ui/road_toolbar_create_menu.gd

```diff
@@ -87,65 +99,106 @@ func on_toolbar_show(primary_sel: Node) -> void:
 
 	pup.add_separator()
 	idx += 1
+
 	pup.add_icon_item(ICN_CT, "RoadContainer", CreateMenu.CONTAINER)
 	pup.set_item_icon_max_width(idx, width)
 	pup.set_item_tooltip(idx, "Adds a RoadContianer child to RoadManager")
 	idx += 1
+
 	pup.add_icon_item(ICN_RP, "RoadPoint", CreateMenu.POINT)
 	pup.set_item_icon_max_width(idx, width)
 	pup.set_item_tooltip(idx, "Adds a new RoadPoint")
 	idx += 1
+
 	pup.add_icon_item(ICN_LN, "RoadLane (AI path)", CreateMenu.LANE)
 	pup.set_item_icon_max_width(idx, width)
 	pup.set_item_tooltip(idx, "Adds a RoadLane which can be used for AI paths")
 	idx += 1
+
 	pup.add_icon_item(ICN_AG, "RoadLaneAgent (AI)", CreateMenu.LANEAGENT)
 	pup.set_item_icon_max_width(idx, width)
 	pup.set_item_tooltip(idx, "Adds a RoadLaneAgent to follow RoadLane paths")
 	idx += 1
+
 	pup.add_separator()
 	idx += 1
+
 	pup.add_item("2x2 road", CreateMenu.TWO_X_TWO)
 	pup.set_item_tooltip(idx, "Adds a segment of road with 2 lanes each way")
 	idx += 1
+
 	# rc_items must be name of the child of this menu
 	if not is_instance_valid(rc_submenu):
 		load_submenu()
+
 	rc_submenu.name = "rc_items"
 	pup.add_submenu_item("RoadContainer presets", "rc_items", idx)
 	idx += 1
-	
+
+	pup.add_separator("Lock new node rotation?")
+	idx += 1
+
+	pup.add_check_item("Lock X", CreateMenu.LOCK_ROTATION_X)
+	pup.set_item_checked(idx, x_rotation_locked)
+	idx += 1
+
+	pup.add_check_item("Lock Y", CreateMenu.LOCK_ROTATION_Y)
+	pup.set_item_checked(idx, y_rotation_locked)
+	idx += 1
+
+	pup.add_check_item("Lock Z", CreateMenu.LOCK_ROTATION_Z)
+	pup.set_item_checked(idx, z_rotation_locked)
+	idx += 1
+
 	pup.add_separator()
-	pup.add_item("Export RoadContainer", CreateMenu.EXPORT_MESH)
 	idx += 1
+
+	pup.add_item("Export RoadContainer", CreateMenu.EXPORT_MESH)
```

Comment:
Maybe an existing bug, but I noticed that having this above the previous `idx += 1` it would not properly disable